diff -r -C 3 A.indent-2.2.12/man/Makefile.in B.indent-2.2.12/man/Makefile.in
*** A.indent-2.2.12/man/Makefile.in     2018-09-06 06:00:11.000000000 +0900
--- B.indent-2.2.12/man/Makefile.in     2018-11-20 13:54:09.714655065 +0900
***************
*** 178,184 ****
  CATOBJEXT = @CATOBJEXT@
  CC = @CC@
  CCDEPMODE = @CCDEPMODE@
! CC_FOR_BUILD = @CC_FOR_BUILD@
  CFLAGS = @CFLAGS@
  CFLAG_VISIBILITY = @CFLAG_VISIBILITY@
  CPP = @CPP@
--- 178,184 ----
  CATOBJEXT = @CATOBJEXT@
  CC = @CC@
  CCDEPMODE = @CCDEPMODE@
! CC_FOR_BUILD = @CC@
  CFLAGS = @CFLAGS@
  CFLAG_VISIBILITY = @CFLAG_VISIBILITY@
  CPP = @CPP@
