 src/gfx_fonts.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gfx_fonts.cpp b/src/gfx_fonts.cpp
index b00816e..2cedd8a 100644
--- a/src/gfx_fonts.cpp
+++ b/src/gfx_fonts.cpp
@@ -293,10 +293,10 @@ void set_font_color(int no, int r, int g, int b)
  */
 void setup_font(TTF_Font *font)
 {
-  char *familyname = TTF_FontFaceFamilyName(font);
+  const char *familyname = TTF_FontFaceFamilyName(font);
   if(familyname)
     log_info("The family name of the face in the font is: %s", familyname);
-  char *stylename = TTF_FontFaceStyleName(font);
+  const char *stylename = TTF_FontFaceStyleName(font);
   if(stylename)
     log_info("The name of the face in the font is: %s", stylename);
   log_info("The font max height is: %d", TTF_FontHeight(font));
