--- DevIL/src-IL/src/il_jp2.cpp.orig	2021-04-16 07:41:49.470068238 -0400
+++ DevIL/src-IL/src/il_jp2.cpp	2021-04-16 07:42:29.726087509 -0400
@@ -314,13 +314,13 @@
 
 
 
-static int iJp2_file_read(jas_stream_obj_t *obj, char *buf, int cnt)
+static int iJp2_file_read(jas_stream_obj_t *obj, char *buf, unsigned int cnt)
 {
 	obj;
 	return iread(buf, 1, cnt);
 }
 
-static int iJp2_file_write(jas_stream_obj_t *obj, char *buf, int cnt)
+static int iJp2_file_write(jas_stream_obj_t *obj, const char *buf, unsigned int cnt)
 {
 	obj;
 	return iwrite(buf, 1, cnt);
