--- ctcs.orig	2008-04-26 22:18:06.000000000 +0200
+++ ctcs	2009-02-27 01:50:17.000000000 +0100
@@ -105,2 +105,3 @@
 my $port = 2780;
+my $listen = "127.0.0.1";
 my $proto = getprotobyname('tcp');
@@ -113,2 +114,3 @@
 	elsif (/^-i$/) { $bwinterval = shift(); }
+	elsif (/^-l$/) { $listen = shift(); }
 	elsif (/^-p$/) { $port = shift(); }
@@ -135,3 +137,3 @@
 	pack("l", 1)) 	|| die "setsockopt: $!";
-bind(Server, sockaddr_in($port, INADDR_ANY))	|| die "bind: $!";
+bind(Server, sockaddr_in($port, inet_aton($listen)))	|| die "bind: $!";
 listen(Server,SOMAXCONN) 				|| die "listen: $!";
@@ -2475,3 +2477,3 @@
 sub usage {
-	print "Usage:  $0 [-d <dlimit>] [-u <ulimit>] [-i <interval>] [-p <port>] [-P]\r\n";
+	print "Usage:  $0 [-d <dlimit>] [-u <ulimit>] [-i <interval>] [-l <address>] [-p <port>] [-P]\r\n";
 }
