diff --git a/Plugins/StreamLinesRepresentation/CMakeLists.txt b/Plugins/StreamLinesRepresentation/CMakeLists.txt
index 5d656a54b830b120c9fc06a06fb337463963bc38..86c24964fe1c8b848b1b7664d673906b2dbd86a6 100644
--- a/Plugins/StreamLinesRepresentation/CMakeLists.txt
+++ b/Plugins/StreamLinesRepresentation/CMakeLists.txt
@@ -41,7 +41,6 @@ encode_files_as_strings(ENCODED_STRING_FILES
 
 add_paraview_plugin(
   StreamLinesRepresentation "0.1"
-  DOCUMENTATION_DIR "${CMAKE_CURRENT_SOURCE_DIR}/doc"
   SERVER_MANAGER_XML StreamLinesRepresentation.xml
   SERVER_MANAGER_SOURCES
     vtkStreamLinesRepresentation.cxx
