--- ./src/crypto.c	2020-03-04 21:50:10.439160670 +0100
+++ ./src/crypto.c	2020-03-04 21:53:10.078241883 +0100
@@ -415,7 +415,7 @@
 {
 	unsigned char key[EVP_MAX_KEY_LENGTH], iv[EVP_MAX_IV_LENGTH];
 	unsigned int out_cumul_len = 0;
-	EVP_CIPHER_CTX ctx;
+	EVP_CIPHER_CTX* ctx = EVP_CIPHER_CTX_new();
 	int out_len = 0;
 	hxmc_t *out;
 
@@ -425,17 +425,17 @@
 	    1, key, iv) <= 0)
 		return EHD_KEYDEC_OTHER;
 
-	out = HXmc_meminit(NULL, par->d_keysize + par->s_cipher->block_size);
-	EVP_CIPHER_CTX_init(&ctx);
-	EVP_DecryptInit_ex(&ctx, par->s_cipher, NULL, key, iv);
-	EVP_DecryptUpdate(&ctx, signed_cast(unsigned char *,
+	out = HXmc_meminit(NULL, par->d_keysize + EVP_CIPHER_block_size(par->s_cipher));
+	EVP_CIPHER_CTX_init(ctx);
+	EVP_DecryptInit_ex(ctx, par->s_cipher, NULL, key, iv);
+	EVP_DecryptUpdate(ctx, signed_cast(unsigned char *,
 		&out[out_len]), &out_len, par->d_text, par->d_keysize);
 	out_cumul_len += out_len;
-	EVP_DecryptFinal_ex(&ctx, signed_cast(unsigned char *,
+	EVP_DecryptFinal_ex(ctx, signed_cast(unsigned char *,
 		&out[out_len]), &out_len);
 	out_cumul_len += out_len;
 	HXmc_setlen(&out, out_cumul_len);
-	EVP_CIPHER_CTX_cleanup(&ctx);
+	EVP_CIPHER_CTX_cleanup(ctx);
 
 	par->d_result = out;
 	return EHD_KEYDEC_SUCCESS;
