--- a/src/VBox/HostDrivers/Support/linux/Makefile
+++ b/src/VBox/HostDrivers/Support/linux/Makefile
@@ -133,7 +133,7 @@ ifdef VBOX_WITH_NATIVE_DTRACE
  MOD_OBJS += SUPDrvDTrace.o
 endif

-MOD_INCL  = $(addprefix -I$(KBUILD_EXTMOD),/ /include /r0drv/linux)
+MOD_INCL  = $(addprefix -I$(KBUILD_EXTMOD)/vboxdrv,/ /include /r0drv/linux)
 ifdef VBOX_WITH_NATIVE_DTRACE
  MOD_INCL += -I/usr/include/linux -I/usr/include
 endif
@@ -159,7 +159,7 @@ ifdef VBOX_WITH_TEXT_MODMEM_HACK
 endif

 # build defs
-MOD_CFLAGS = -include $(KBUILD_EXTMOD)/include/VBox/SUPDrvMangling.h \
+MOD_CFLAGS = -include $(KBUILD_EXTMOD)/vboxdrv/include/VBox/SUPDrvMangling.h \
 	-fno-omit-frame-pointer -fno-pie

 include $(obj)/Makefile.include.footer
--- a/src/VBox/HostDrivers/VBoxNetAdp/linux/Makefile
+++ b/src/VBox/HostDrivers/VBoxNetAdp/linux/Makefile
@@ -42,7 +42,7 @@ MOD_OBJS += math/gcc/divdi3.o \
 	math/gcc/umoddi3.o
 endif

-MOD_INCL = $(addprefix -I$(KBUILD_EXTMOD),/ /include /r0drv/linux)
+MOD_INCL = $(addprefix -I$(KBUILD_EXTMOD)/vboxnetadp,/ /include /r0drv/linux)
 MOD_DEFS = -DRT_OS_LINUX -DIN_RING0 -DIN_RT_R0 -DIN_SUP_R0 -DVBOX \
 	    -DRT_WITH_VBOX -DVBOX_WITH_HARDENING \
 	    -Wno-declaration-after-statement
@@ -67,6 +67,6 @@ ifdef VBOX_USE_INSERT_PAGE
 endif

 # build defs
-MOD_CFLAGS = -include $(KBUILD_EXTMOD)/include/VBox/SUPDrvMangling.h -fno-pie
+MOD_CFLAGS = -include $(KBUILD_EXTMOD)/vboxnetadp/include/VBox/SUPDrvMangling.h -fno-pie

 include $(obj)/Makefile.include.footer
--- a/src/VBox/HostDrivers/VBoxNetFlt/linux/Makefile
+++ b/src/VBox/HostDrivers/VBoxNetFlt/linux/Makefile
@@ -46,7 +46,7 @@ MOD_OBJS += math/gcc/divdi3.o \
 	math/gcc/umoddi3.o
 endif

-MOD_INCL = $(addprefix -I$(KBUILD_EXTMOD),/ /include /r0drv/linux)
+MOD_INCL = $(addprefix -I$(KBUILD_EXTMOD)/vboxnetflt,/ /include /r0drv/linux)
 MOD_DEFS = -DRT_OS_LINUX -DIN_RING0 -DIN_RT_R0 \
 	-DIN_SUP_R0 -DVBOX -DRT_WITH_VBOX -DVBOX_WITH_HARDENING \
 	-Wno-declaration-after-statement
@@ -71,6 +71,6 @@ ifdef VBOX_USE_INSERT_PAGE
 endif

 # build defs
-MOD_CFLAGS = -include $(KBUILD_EXTMOD)/include/VBox/SUPDrvMangling.h -fno-pie
+MOD_CFLAGS = -include $(KBUILD_EXTMOD)/vboxnetflt/include/VBox/SUPDrvMangling.h -fno-pie

 include $(obj)/Makefile.include.footer
--- a/src/VBox/HostDrivers/VBoxPci/linux/Makefile
+++ b/src/VBox/HostDrivers/VBoxPci/linux/Makefile
@@ -46,7 +46,7 @@ MOD_OBJS += math/gcc/divdi3.o       \
 	math/gcc/umoddi3.o
 endif

-MOD_INCL = $(addprefix -I$(KBUILD_EXTMOD),/ /include /r0drv/linux)
+MOD_INCL = $(addprefix -I$(KBUILD_EXTMOD)/vboxpci,/ /include /r0drv/linux)
 MOD_DEFS = -DRT_OS_LINUX -DIN_RING0 -DIN_RT_R0 -DIN_SUP_R0 -DVBOX \
 	-DRT_WITH_VBOX -DVBOX_WITH_HARDENING
 ifeq ($(BUILD_TARGET_ARCH),amd64)
@@ -68,6 +68,6 @@ ifdef VBOX_USE_INSERT_PAGE
 endif

 # build defs
-MOD_CFLAGS = -include $(KBUILD_EXTMOD)/include/VBox/SUPDrvMangling.h -fno-pie
+MOD_CFLAGS = -include $(KBUILD_EXTMOD)/vboxpci/include/VBox/SUPDrvMangling.h -fno-pie

 include $(obj)/Makefile.include.footer
