--- Makefile.am	2022-02-11 12:32:48.389406522 +0100
+++ Makefile.am	2022-02-11 12:33:26.063313541 +0100
@@ -713,8 +713,8 @@
 		  echo hpaio >>$(DESTDIR)/etc/sane.d/dll.conf ; \
 	   fi; \
  	   if [ \( "$(UNAME)" = "x86_64" -a  -d "$(libdir)/x86_64-linux-gnu/sane" \) ]; then \
-	  	ln -sf $(libdir)/sane/libsane-hpaio.so $(libdir)/x86_64-linux-gnu/sane/ ; \
-	  	ln -sf $(libdir)/sane/libsane-hpaio.so.1 $(libdir)/x86_64-linux-gnu/sane/ ; \
+	  	ln -sf $(libdir)/x86_64-linux-gnu/sane/libsane-hpaio.so.1.0.0 $(libdir)/sane/libsane-hpaio.so.1 ; \
+	  	ln -sf $(libdir)/x86_64-linux-gnu/sane/libsane-hpaio.so.1 $(libdir)/sane/libsane-hpaio.so ; \
 	   fi; \
 	   if [ \( \( "$(UNAME)" = "i686" -o "$(UNAME)" = "i386" \) -a -d "$(libdir)/i386-linux-gnu" \) ]; then \
 	      ln -sf $(libdir)/libhpmud.so.0.0.6  $(libdir)/i386-linux-gnu/libhpmud.so ; \
