diff --git a/hp-uiscan.desktop.in b/hp-uiscan.desktop.in
index 4a551e5..3f99547 100644
--- a/hp-uiscan.desktop.in
+++ b/hp-uiscan.desktop.in
@@ -5,4 +5,7 @@ Type=Application
 Terminal=false
 Name=hp-uiscan
 Exec=/usr/bin/hp-uiscan
-Icon=/usr/share/icons/Humanity/devices/48/printer.svg
+Icon=printer
+Categories=Utility;
+GenericName=HP Linux Imaging and Printing System
+Comment=SANE-based scan utility for HPLIP supported all-in-one/mfp devices.
