--- ./meson.build.orig	2020-09-22 20:55:26.000000000 +0300
+++ ./meson.build	    2021-03-27 17:48:04.098297969 +0200
@@ -272,8 +272,8 @@
 
 # Brotli decompression support
 brotli_dep = dependency('libbrotlidec', required: get_option('brotli'))
-ftoption_command += ['--enable=FT_CONFIG_OPTION_USE_BROTLI']
-ft2_deps += [brotli_dep]
+#ftoption_command += ['--enable=FT_CONFIG_OPTION_USE_BROTLI']
+#ft2_deps += [brotli_dep]
 
 # We can now generate `ftoption.h`.
 ftoption_h = custom_target('ftoption.h',
