diff -r -C 2 efitools-1.9.2/cert-to-efi-sig-list.c efitools-1.9.2.patched/cert-to-efi-sig-list.c
*** efitools-1.9.2/cert-to-efi-sig-list.c	2023-01-25 17:33:20.000000000 +0900
--- efitools-1.9.2.patched/cert-to-efi-sig-list.c	2025-10-29 15:10:12.473183816 +0900
***************
*** 7,11 ****
--- 7,13 ----
  
  #include <stdint.h>
+ #ifndef __STDC_VERSION__
  #define __STDC_VERSION__ 199901L
+ #endif
  #include <efi.h>
  #ifdef CONFIG_arm
diff -r -C 2 efitools-1.9.2/efi-keytool.c efitools-1.9.2.patched/efi-keytool.c
*** efitools-1.9.2/efi-keytool.c	2023-01-25 17:33:20.000000000 +0900
--- efitools-1.9.2.patched/efi-keytool.c	2025-10-29 15:10:22.599968136 +0900
***************
*** 16,20 ****
--- 16,23 ----
  #include <unistd.h>
  
+ #ifndef __STDC_VERSION__
  #define __STDC_VERSION__ 199901L
+ #endif
+ 
  #include <efi.h>
  
diff -r -C 2 efitools-1.9.2/efi-readvar.c efitools-1.9.2.patched/efi-readvar.c
*** efitools-1.9.2/efi-readvar.c	2023-01-25 17:33:20.000000000 +0900
--- efitools-1.9.2.patched/efi-readvar.c	2025-10-29 15:10:29.120044118 +0900
***************
*** 18,22 ****
--- 18,24 ----
  #include <openssl/x509.h>
  
+ #ifndef __STDC_VERSION__
  #define __STDC_VERSION__ 199901L
+ #endif
  #include <efi.h>
  
diff -r -C 2 efitools-1.9.2/efi-updatevar.c efitools-1.9.2.patched/efi-updatevar.c
*** efitools-1.9.2/efi-updatevar.c	2023-01-25 17:33:20.000000000 +0900
--- efitools-1.9.2.patched/efi-updatevar.c	2025-10-29 15:23:38.073418679 +0900
***************
*** 21,25 ****
--- 21,27 ----
  #include <openssl/pem.h>
  
+ #ifndef __STDC_VERSION__
  #define __STDC_VERSION__ 199901L
+ #endif
  #include <efi.h>
  
diff -r -C 2 efitools-1.9.2/flash-var.c efitools-1.9.2.patched/flash-var.c
*** efitools-1.9.2/flash-var.c	2023-01-25 17:33:20.000000000 +0900
--- efitools-1.9.2.patched/flash-var.c	2025-10-29 15:24:20.310650700 +0900
***************
*** 1,2 ****
--- 1,6 ----
+ #ifndef _XOPEN_SOURCE
+ #define _XOPEN_SOURCE
+ #endif
+ 
  #include <stdlib.h>
  #include <stdint.h>
***************
*** 11,15 ****
--- 15,21 ----
  #include <unistd.h>
  
+ #ifndef __STDC_VERSION__
  #define __STDC_VERSION__ 199901L
+ #endif
  #include <efi.h>
  
diff -r -C 2 efitools-1.9.2/hash-to-efi-sig-list.c efitools-1.9.2.patched/hash-to-efi-sig-list.c
*** efitools-1.9.2/hash-to-efi-sig-list.c	2023-01-25 17:33:20.000000000 +0900
--- efitools-1.9.2.patched/hash-to-efi-sig-list.c	2025-10-29 15:23:54.153633733 +0900
***************
*** 5,9 ****
--- 5,11 ----
   */
  #include <stdint.h>
+ #ifndef __STDC_VERSION__
  #define __STDC_VERSION__ 199901L
+ #endif
  #include <efi.h>
  #ifdef CONFIG_arm
diff -r -C 2 efitools-1.9.2/sig-list-to-certs.c efitools-1.9.2.patched/sig-list-to-certs.c
*** efitools-1.9.2/sig-list-to-certs.c	2023-01-25 17:33:20.000000000 +0900
--- efitools-1.9.2.patched/sig-list-to-certs.c	2025-10-29 15:23:59.643707209 +0900
***************
*** 5,9 ****
--- 5,11 ----
   */
  #include <stdint.h>
+ #ifndef __STDC_VERSION__
  #define __STDC_VERSION__ 199901L
+ #endif
  #include <efi.h>
  #ifdef CONFIG_arm
diff -r -C 2 efitools-1.9.2/sign-efi-sig-list.c efitools-1.9.2.patched/sign-efi-sig-list.c
*** efitools-1.9.2/sign-efi-sig-list.c	2023-01-25 17:33:20.000000000 +0900
--- efitools-1.9.2.patched/sign-efi-sig-list.c	2025-10-29 15:09:30.496034335 +0900
***************
*** 5,9 ****
--- 5,16 ----
   */
  #include <stdint.h>
+ #ifndef __STDC_VERSION__
  #define __STDC_VERSION__ 199901L
+ #endif
+ 
+ #ifndef _XOPEN_SOURCE
+ #define _XOPEN_SOURCE
+ #endif
+ 
  #include <efi.h>
  #ifdef CONFIG_arm
