--- kernel-4.0.2/Makefile.linux.orig	2002-07-27 23:54:25.000000000 -0400
+++ kernel-4.0.2/Makefile.linux	2002-07-27 23:57:51.000000000 -0400
@@ -47,19 +47,13 @@
 
 # **** End of SMP/MODVERSIONS detection
 
-MODS=           gamma.o tdfx.o r128.o
+MODS=           r128.o
 LIBS=           libdrm.a
 
 DRMOBJS=	init.o memory.o proc.o auth.o context.o drawable.o bufs.o \
 		lists.o lock.o ioctl.o fops.o vm.o dma.o ctxbitmap.o
 DRMHEADERS=	drm.h drmP.h compat-pre24.h
 
-GAMMAOBJS=      gamma_drv.o gamma_dma.o
-GAMMAHEADERS=   gamma_drv.h $(DRMHEADERS)
-
-TDFXOBJS=       tdfx_drv.o tdfx_context.o
-TDFXHEADERS=    tdfx_drv.h $(DRMHEADERS)
-
 R128OBJS=	r128_drv.o r128_cce.o r128_bufs.o r128_state.o r128_context.o
 R128HEADERS=	r128_drv.h r128_drm.h $(DRMHEADERS)
 
@@ -132,17 +126,6 @@
 ifeq ($(AGP),1)
 MODCFLAGS += -DCONFIG_AGP -DCONFIG_AGP_MODULE
 DRMOBJS += agpsupport.o
-MODS += mga.o
-ifeq ($(MACHINE),i386)
-MODS += i810.o
-endif
-
-
-MGAOBJS=	mga_drv.o mga_dma.o mga_bufs.o mga_state.o mga_context.o
-MGAHEADERS=	mga_drv.h $(DRMHEADERS)
-
-I810OBJS=	i810_drv.o i810_dma.o i810_bufs.o i810_context.o
-I810HEADERS=	i810_drv.h $(DRMHEADERS)
 endif
 
 ifeq ($(SIS),1)
@@ -192,16 +175,6 @@
 	-$(RM) -f $@
 	$(AR) rcs $@ $(DRMOBJS)
 
-gamma_drv.o: gamma_drv.c
-	$(CC) $(MODCFLAGS) -DEXPORT_SYMTAB -I$(TREE) -c $< -o $@
-gamma.o: $(GAMMAOBJS) $(LIBS)
-	$(LD) -r $^ -o $@
-
-tdfx_drv.o: tdfx_drv.c
-	$(CC) $(MODCFLAGS) -DEXPORT_SYMTAB -I$(TREE) -c $< -o $@
-tdfx.o: $(TDFXOBJS) $(LIBS)
-	$(LD) -r $^ -o $@
-
 r128_drv.o: r128_drv.c
 	$(CC) $(MODCFLAGS) -DEXPORT_SYMTAB -I$(TREE) -c $< -o $@
 r128.o: $(R128OBJS) $(LIBS)
@@ -210,18 +183,6 @@
 sis.o: $(SISOBJS) $(LIBS)
 	$(LD) -r $^ -o $@
 
-ifeq ($(AGP),1)
-mga_drv.o: mga_drv.c
-	$(CC) $(MODCFLAGS) -DEXPORT_SYMTAB -I$(TREE) -c $< -o $@
-mga.o: $(MGAOBJS) $(LIBS)
-	$(LD) -r $^ -o $@
-
-i810_drv.o: i810_drv.c
-	$(CC) $(MODCFLAGS) -DEXPORT_SYMTAB -I$(TREE) -c $< -o $@
-i810.o: $(I810OBJS) $(LIBS)
-	$(LD) -r $^ -o $@
-endif
-
 .PHONY: ChangeLog
 ChangeLog:
 	@rm -f Changelog
@@ -236,13 +197,7 @@
 	$(CC) $(MODCFLAGS) -I$(TREE) -c $< -o $@
 
 $(DRMOBJS): $(DRMHEADERS)
-$(GAMMAOBJS): $(GAMMAHEADERS)
-$(TDFXOBJS): $(TDFXHEADERS)
 $(R128OBJS): $(R128HEADERS)
-ifeq ($(AGP),1)
-$(MGAOBJS): $(MGAHEADERS)
-$(I810OBJS): $(I810HEADERS)
-endif
 
 clean:
 	rm -f *.o *.a *~ core
