--- ./configure.orig	2019-10-23 00:41:54.000000000 +0300
+++ ./configure	2019-10-26 05:44:56.623221843 +0300
@@ -43317,8 +43317,8 @@
 $as_echo "$ac_cv_lib_gss_gss_import_name" >&6; }
 if test "x$ac_cv_lib_gss_gss_import_name" = xyes; then :
   GSSAPI_LIBS="-lgss"
-else
-  as_fn_error $? "cannot find usable GSS-API library" "$LINENO" 5
+#else
+#  as_fn_error $? "cannot find usable GSS-API library" "$LINENO" 5
 fi
 
 fi
@@ -44184,8 +44184,8 @@
 $as_echo "$ac_cv_lib_gss_gss_import_name" >&6; }
 if test "x$ac_cv_lib_gss_gss_import_name" = xyes; then :
   GSSAPI_LIBS="-lgss"
-else
-  as_fn_error $? "cannot find usable GSS-API library" "$LINENO" 5
+#else
+#  as_fn_error $? "cannot find usable GSS-API library" "$LINENO" 5
 fi
 
 fi
@@ -44976,8 +44976,8 @@
 $as_echo "$ac_cv_lib_gss_gss_import_name" >&6; }
 if test "x$ac_cv_lib_gss_gss_import_name" = xyes; then :
   GSSAPI_LIBS="-lgss"
-else
-  as_fn_error $? "cannot find usable GSS-API library" "$LINENO" 5
+#else
+#  as_fn_error $? "cannot find usable GSS-API library" "$LINENO" 5
 fi
 
 fi
@@ -45767,8 +45767,8 @@
 $as_echo "$ac_cv_lib_gss_gss_import_name" >&6; }
 if test "x$ac_cv_lib_gss_gss_import_name" = xyes; then :
   GSSAPI_LIBS="-lgss"
-else
-  as_fn_error $? "cannot find usable GSS-API library" "$LINENO" 5
+#else
+#  as_fn_error $? "cannot find usable GSS-API library" "$LINENO" 5
 fi
 
 fi
