--- xearth-1.1/markers.c.orig	2002-06-08 11:24:21.000000000 -0400
+++ xearth-1.1/markers.c	2002-06-08 11:24:10.000000000 -0400
@@ -83,6 +83,7 @@
   "  31.05   31.25 \"Cairo\"               # Egypt                 ",
   "  22.5    88.3  \"Calcutta\"            # India                 ",
   " -33.93   18.47 \"Cape Town\"           # South Africa          ",
+  "  35.82  -78.72 \"Cary\"                # North Carolina, USA   ",
   "  33.6    -7.6  \"Casablanca\"          # Morocco (Rabat?)      ",
   "  41.83  -87.75 \"Chicago\"             # Illinois, USA         ",
   "  32.78  -96.80 \"Dallas\"              # Texas, USA            ",
