From e22ddac4496ce68704012d64cd0b5a2b189306ad Mon Sep 17 00:00:00 2001
From: Guido Cella <guidocella91@gmail.com>
Date: Sun, 18 Oct 2020 07:10:03 +0200
Subject: [PATCH] Center floating windows

---
 dwl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dwl.c b/dwl.c
index 730e46a..b9449ad 100644
--- a/dwl.c
+++ b/dwl.c
@@ -476,6 +476,8 @@ applyrules(Client *c)
 					mon = m;
 		}
 	}
+	c->geom.x = (mon->w.width - c->geom.width) / 2;
+	c->geom.y = (mon->w.height - c->geom.height) / 2;
 	setmon(c, mon, newtags);
 }
 
