From 6794299c06659fe6623b3a4b03a8c6102d7efe8a Mon Sep 17 00:00:00 2001
From: sam-barr <mail@samf.bar>
Date: Sat, 16 Jan 2021 13:56:36 -0500
Subject: [PATCH 1/2] Implement attach above

---
 dwl.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/dwl.c b/dwl.c
index c09f598..65010dc 100644
--- a/dwl.c
+++ b/dwl.c
@@ -1345,10 +1345,16 @@ mapnotify(struct wl_listener *listener, void *data)
 		return;
 	}
 
-	/* Insert this client into client lists. */
-	wl_list_insert(&clients, &c->link);
-	wl_list_insert(&fstack, &c->flink);
-	wl_list_insert(&stack, &c->slink);
+        /* Insert this client into client lists. */
+        if (oldfocus) {
+                wl_list_insert(oldfocus->link.prev, &c->link);
+                wl_list_insert(oldfocus->flink.prev, &c->flink);
+                wl_list_insert(oldfocus->slink.prev, &c->slink);
+        } else {
+                wl_list_insert(&clients, &c->link);
+                wl_list_insert(&fstack, &c->flink);
+                wl_list_insert(&stack, &c->slink);
+        }
 
 	client_get_geometry(c, &c->geom);
 	c->geom.width += 2 * c->bw;

From 25661d4b43e2078fcac7c49a8fcb33e49ef4a71b Mon Sep 17 00:00:00 2001
From: sam-barr <mail@samf.bar>
Date: Sat, 16 Jan 2021 14:03:18 -0500
Subject: [PATCH 2/2] replaced spaces with tabs

---
 dwl.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/dwl.c b/dwl.c
index 65010dc..b7541d6 100644
--- a/dwl.c
+++ b/dwl.c
@@ -1345,16 +1345,16 @@ mapnotify(struct wl_listener *listener, void *data)
 		return;
 	}
 
-        /* Insert this client into client lists. */
-        if (oldfocus) {
-                wl_list_insert(oldfocus->link.prev, &c->link);
-                wl_list_insert(oldfocus->flink.prev, &c->flink);
-                wl_list_insert(oldfocus->slink.prev, &c->slink);
-        } else {
-                wl_list_insert(&clients, &c->link);
-                wl_list_insert(&fstack, &c->flink);
-                wl_list_insert(&stack, &c->slink);
-        }
+	/* Insert this client into client lists. */
+	if (oldfocus) {
+		wl_list_insert(oldfocus->link.prev, &c->link);
+		wl_list_insert(oldfocus->flink.prev, &c->flink);
+		wl_list_insert(oldfocus->slink.prev, &c->slink);
+	} else {
+		wl_list_insert(&clients, &c->link);
+		wl_list_insert(&fstack, &c->flink);
+		wl_list_insert(&stack, &c->slink);
+	}
 
 	client_get_geometry(c, &c->geom);
 	c->geom.width += 2 * c->bw;
