--- ./rts/Game/ChatMessage.h	2021-01-01 18:26:15.000000000 +0100
+++ ./rts/Game/ChatMessage.h.new	2026-01-05 23:06:07.310247939 +0100
@@ -5,6 +5,7 @@
 
 #include <memory>
 #include <string>
+#include <cstdint>
 
 namespace netcode {
 	class RawPacket;
--- ./rts/ExternalAI/SkirmishAIWrapper.h	2021-01-01 18:26:15.000000000 +0100
+++ ./rts/ExternalAI/SkirmishAIWrapper.h.new	2026-01-05 23:13:48.201704618 +0100
@@ -4,6 +4,7 @@
 #define SKIRMISH_AI_WRAPPER_H
 
 #include "SkirmishAIKey.h"
+#include <cstdint>
 
 class CSkirmishAILibrary;
 struct SSkirmishAICallback;
