--- ./src/prog/gpm-root.y	2012-10-26 16:21:38.000000000 -0500
+++ ./src/prog/gpm-root.y	2017-10-10 13:50:02.115721252 -0500
@@ -1196,11 +1196,7 @@
                                                         LOG_DAEMON : LOG_USER);
    /* reap your zombies */
    childaction.sa_handler=reap_children;
-#if defined(__GLIBC__)
-   __sigemptyset(&childaction.sa_mask);
-#else /* __GLIBC__ */
-   childaction.sa_mask=0;
-#endif /* __GLIBC__ */
+   sigemptyset(&childaction.sa_mask);
    childaction.sa_flags=SA_INTERRUPT; /* need to break the select() call */
    sigaction(SIGCHLD,&childaction,NULL);
 
