--- ./config.h.in.old	2018-08-10 17:46:51.000000000 +0300
+++ ./config.h.in	2018-11-05 09:10:05.504660985 +0200
@@ -70,10 +70,6 @@
    0 if you don't. */
 #undef HAVE_DECL_ASN1_STRING_GET0_DATA
 
-/* Define to 1 if you have the declaration of `BN_get_rfc3526_prime_2048', and
-   to 0 if you don't. */
-#undef HAVE_DECL_BN_GET_RFC3526_PRIME_2048
-
 /* Define to 1 if you have the declaration of `DH_set0_pqg', and to 0 if you
    don't. */
 #undef HAVE_DECL_DH_SET0_PQG
