--- ./dirmngr/dirmngr.c.orig	2021-05-26 19:50:15.000000000 +0300
+++ ./dirmngr/dirmngr.c	2021-06-19 10:28:51.246062962 +0300
@@ -779,7 +779,7 @@
         p = pargs->r.ret_str;
         if (!strncmp (p, "ldap:", 5) && !(p[5] == '/' && p[6] == '/'))
           p += 5;
-
+#if USE_LDAP
         server = ldapserver_parse_one (p, NULL, 0);
         if (server)
           {
@@ -792,6 +792,7 @@
             server->next = opt.ldapservers;
             opt.ldapservers = server;
           }
+#endif
       }
       break;
 
