--- frmts/jpeg2000/jpeg2000_vsil_io.cpp.orig	2020-08-11 04:42:24.216334002 -0400
+++ frmts/jpeg2000/jpeg2000_vsil_io.cpp	2020-08-11 04:45:15.181503955 -0400
@@ -94,13 +94,23 @@
 * File stream object.
 \******************************************************************************/
 
+// PRIjas_seqent macro is defined since Jasper 2.0.17
+
+#ifndef PRIjas_seqent
 static int JPEG2000_VSIL_read(jas_stream_obj_t *obj, char *buf, int cnt)
+#else
+static int JPEG2000_VSIL_read(jas_stream_obj_t *obj, char *buf, unsigned cnt)
+#endif
 {
     jas_stream_VSIFL_t *fileobj = JAS_CAST(jas_stream_VSIFL_t *, obj);
     return static_cast<int>(VSIFReadL(buf, 1, cnt, fileobj->fp));
 }
 
+#ifndef PRIjas_seqent
 static int JPEG2000_VSIL_write(jas_stream_obj_t *obj, char *buf, int cnt)
+#else
+static int JPEG2000_VSIL_write(jas_stream_obj_t *obj, char *buf, unsigned int cnt)
+#endif
 {
     jas_stream_VSIFL_t *fileobj = JAS_CAST(jas_stream_VSIFL_t *, obj);
     return static_cast<int>(VSIFWriteL(buf, 1, cnt, fileobj->fp));
