--- ./docs/tools/shooter.c.old	2018-12-12 19:46:09.000000000 +0200
+++ ./docs/tools/shooter.c	2018-12-21 01:42:45.707794264 +0200
@@ -170,7 +170,7 @@
       y_orig = 0;
     }
 
-  if (x_orig + width > gdk_screen_get_width (gdk_screen_get_dfeault ()))
+  if (x_orig + width > gdk_screen_get_width (gdk_screen_get_default ()))
     width = gdk_screen_width () - x_orig;
 
   if (y_orig + height > gdk_screen_get_height (gdk_screen_get_default ()))
