From 9d261799e616cdc65ccefa06c6072e8158a302c5 Mon Sep 17 00:00:00 2001
From: Marcelo Roberto Jimenez <marcelo.jimenez@gmail.com>
Date: Thu, 25 Apr 2024 18:54:49 -0300
Subject: [PATCH] Fixes broken cmake regex

---
 cmake/autoheader.cmake | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/cmake/autoheader.cmake b/cmake/autoheader.cmake
index df6162257..045b90a50 100644
--- a/cmake/autoheader.cmake
+++ b/cmake/autoheader.cmake
@@ -18,9 +18,11 @@ if (NOT PUPNP_VERSION_STRING)
 			string (REGEX REPLACE ";" "" line ${line})
 			string (REGEX REPLACE "[ \t\r\n] *" " " line ${line})
 
-			if (line MATCHES "AC_INIT.* ([0-9]*\\.[0-9]*\\.[0-9]*).*")
-				message (STATUS "Setting package-version to ${CMAKE_MATCH_1}")
-				set (PUPNP_VERSION_STRING ${CMAKE_MATCH_1} CACHE STRING "Version of the whole package" FORCE)
+			if (line MATCHES  "AC_INIT\\\(([^,]*),([0-9]*\.[0-9]*\.[0-9]*)([^\)]*)\\\)")
+				message (STATUS "Setting package-version to ${CMAKE_MATCH_2}")
+				set (PUPNP_VERSION_STRING ${CMAKE_MATCH_2} CACHE STRING "Version of the whole package" FORCE)
+				#message (STATUS "Extracted package name is ${CMAKE_MATCH_1}")
+				#message (STATUS "Left from parsing: ${CMAKE_MATCH_3}")
 			elseif (line MATCHES "[. \t]*AC_DEFINE_UNQUOTED *\\(([^,]*), *([^,]*), *([^\\)]*)")
 				set (SAVED_MATCH ${CMAKE_MATCH_1})
 
@@ -76,3 +78,4 @@ if (NOT PUPNP_VERSION_STRING)
 		endforeach()
 	endforeach()
 endif()
+message (STATUS "package-version is ${PUPNP_VERSION_STRING}")
