--- ./configure.old	2002-11-27 16:07:12.000000000 -0800
+++ ./configure	2002-11-27 16:08:45.000000000 -0800
@@ -2087,7 +2087,7 @@
   ac_cv_prog_CXXCPP="$CXXCPP"
 ac_ext=C
 # CXXFLAGS is not in ac_cpp because -g, -O, etc. are not valid cpp options.
-ac_cpp='$CXXCPP $CPPFLAGS'
+ac_cpp='$CXXCPP $CPPFLAGS -Wno-deprecated'
 ac_compile='${CXX-g++} -c $CXXFLAGS $CPPFLAGS conftest.$ac_ext 1>&5'
 ac_link='${CXX-g++} -o conftest${ac_exeext} $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
 cross_compiling=$ac_cv_prog_cxx_cross
