diff -r -C 3 XML-AutoWriter-0.4.orig/Makefile.PL XML-AutoWriter-0.4/Makefile.PL
*** XML-AutoWriter-0.4.orig/Makefile.PL 2018-06-28 11:01:51.397797955 +0900
--- XML-AutoWriter-0.4/Makefile.PL      2018-06-28 11:02:02.577814277 +0900
***************
*** 9,15 ****
  build_requires => 'Test';
  build_requires => 'IO::File';

! auto_set_repository;
  auto_manifest;
  auto_install;
  WriteAll;
--- 9,15 ----
  build_requires => 'Test';
  build_requires => 'IO::File';

! auto_set_repository();
  auto_manifest;
  auto_install;
  WriteAll;
