diff -r -U3 include-what-you-use-0.14.orig/iwyu_include_picker.cc include-what-you-use-0.14.patched/iwyu_include_picker.cc
--- include-what-you-use-0.14.orig/iwyu_include_picker.cc	2020-06-26 16:36:56.180343263 +0200
+++ include-what-you-use-0.14.patched/iwyu_include_picker.cc	2020-06-26 17:06:01.476235254 +0200
@@ -914,7 +914,9 @@
   if (status == kCalculating) {   // means there's a cycle in the mapping
     // TODO: Reconsider cycle handling; the include_cycle test fails without
     // this special-casing, but it seems we should handle this more generally.
-    if (key.find("internal/") != string::npos) {
+    if (key.find("internal/") != string::npos
+        || key.find("boost/preprocessor") != string::npos
+        || key.find("boost/python") != string::npos) {
       VERRS(4) << "Ignoring a cyclical mapping involving " << key << "\n";
       return;
     }
