--- poppler-24.12.0/CMakeLists.txt	2024-12-02 16:21:42.000000000 -0500
+++ poppler-24.12.0-new/CMakeLists.txt	2024-12-04 06:32:53.155287468 -0500
@@ -668,12 +668,14 @@
     poppler/Array.h
     poppler/CachedFile.h
     poppler/Catalog.h
+    poppler/CharCodeToUnicode.h
     poppler/CryptoSignBackend.h
     poppler/DateInfo.h
     poppler/Dict.h
     poppler/Error.h
     poppler/FILECacheLoader.h
     poppler/FileSpec.h
+    poppler/FontEncodingTables.h
     poppler/FontInfo.h
     poppler/Form.h
     poppler/Function.h
