--- ./src/murrine_style.c.orig	2018-12-31 16:56:52.782902284 +0200
+++ ./src/murrine_style.c	2018-12-31 17:02:09.131362568 +0200
@@ -2161,6 +2161,7 @@
 		{
 			cairo_t *cr; 
 			cr = murrine_begin_paint (window, area);
+			
 			cairo_translate (cr, x+xos, y+yos);
 			murrine_set_color_rgba (cr, &temp, 0.5);
 			pango_cairo_show_layout (cr, layout);
@@ -2171,6 +2172,7 @@
 			cairo_t *cr; 
 			cr = murrine_begin_paint (window, area); 
 			cairo_translate (cr, x+xos, y+yos); 
+			pango_cairo_update_layout (cr, layout);
 			pango_cairo_layout_path (cr, layout);
 			murrine_set_color_rgba (cr, &temp, 0.5);
 			cairo_stroke (cr);
